src/wagtail_ai/ai/echo.py

```diff
@@ -0,0 +1,61 @@
+import random
+import time
+from collections.abc import Generator, Iterator
+from typing import Any, NotRequired, TypedDict
+
+from .base import AIBackend, AIResponse, ChatModelConfig
+
+
+class EchoResponse(AIResponse):
+    _text: str | None = None
+    response_iterator: Iterator[str]
+
+    def __init__(self, response_iterator: Iterator[str]) -> None:
+        self.response_iterator = response_iterator
+
+    def __iter__(self) -> Iterator[str]:
+        return self.response_iterator
+
+    def text(self) -> str:
+        if self._text is not None:
+            return self._text
+        self._text = " ".join(self.response_iterator)
+        return self._text
+
+
+class EchoBackendConfig(TypedDict):
+    MAX_WORD_SLEEP_SECONDS: NotRequired[int]
+
+
+class EchoBackend(AIBackend[EchoBackendConfig, ChatModelConfig]):
+    max_word_sleep_seconds: int = 1
```

Comment:
**Question:** Should this default to 0 so we don't need to override it in tests (which I suppose is the main use case for this backend)?

Reply:
Agree.